lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -45,3 +47,51 @@ data class TransactionFailureReason(
     val code: String,
     val errorMessage: String,
 )
+
+@OptIn(ExperimentalSerializationApi::class)
+@Serializable
+data class EstimateFeeResponse(
+    @JsonNames("gas_consumed", "gas_usage")
+    val gasConsumed: Felt,
+
+    @JsonNames("gas_price")
+    val gasPrice: Felt,
+
+    @JsonNames("overall_fee")
+    val overallFee: Felt,
+)
+
+@OptIn(ExperimentalSerializationApi::class)
```

Comment:
Why exactly is a custom serializer needed for this class?

By the way, if you just want to modify the json object for serialization, you can use JsonTransformingSerializer, it's a little easier to use then full fledged custom serializers.

Reply:
Needed because of conversions from numeric values (not necessarily ints or longs) to felts.